Document perfmap settings (#38538)
dotnet/docs: docs/core/runtime-config/debugging-profiling.md

| **Environment variable** | `CORECLR_PROFILER_PATH_32` | *string-path* |
| **Environment variable** | `CORECLR_PROFILER_PATH_64` | *string-path* |

## Export perf maps and jit dumps

- Enables or disables selective enablement of perf maps or jit dumps. These files allow third party tools, such as the Linux `perf` tool, to identify call sites for dynamically generated code and precompiled ReadyToRun (R2R) modules.
- If you omit this setting, writing perf map and jit dump files are both disabled. This is equivalent to setting the value to `0`.
- When perf maps are disabled, not all managed callsites will be properly resolved.
- Depending on the Linux kernel version, both formats are supported by the `perf` tool.

> - Enabling perf maps or jit dumps causes a 10-20% overhead. It is recommended to selectively enable either perf maps or jit dumps, but not both to minimize performance impact.

Reviewer: ```suggestion
- Enabling perf maps or jit dumps causes a 10-20% overhead. To minimize performance impact, it's recommended to selectively enable either perf maps or jit dumps, but not both.
```

> | Format | Description | Supported on |

Reviewer: ```suggestion
The following table compares perf maps and jit maps.

| Format | Description | Supported on |
```

| - | - | - |

> | *Perf maps* | Emits `/tmp/perf-<pid>.map` which contains symbolic information for dynamically generated code.<br/>Emits `/tmp/perfinfo-<pid>.map` which includes ReadyToRun (R2R) module symbol information and is used by [PerfCollect](../diagnostics/trace-perfcollect-lttng.md). | Perf maps are supported on all Linux kernel versions.|

Reviewer: ```suggestion
| *Perf maps* | Emits `/tmp/perf-<pid>.map`, which contains symbolic information for dynamically generated code.<br/>Emits `/tmp/perfinfo-<pid>.map`, which includes ReadyToRun (R2R) module symbol information and is used by [PerfCollect](../diagnostics/trace-perfcollect-lttng.md). | Perf maps are supported on all Linux kernel versions. |
```

> | *Jit dumps* | The jit dump format supercedes perf maps and contains more detailed symbolic information. When enabled, jit dumps will be output to `/tmp/jit-<pid>.dump` files.|Linux kernel versions 5.4 or higher|

Reviewer: ```suggestion
| *Jit dumps* | The jit dump format supersedes perf maps and contains more detailed symbolic information. When enabled, jit dumps are output to `/tmp/jit-<pid>.dump` files. | Linux kernel versions 5.4 or higher. |
```
